### PR TITLE
Prepare components for a flux aggregator

### PR DESCRIPTION
**Pull request by hellobart**
- combined visualisation components to vis component
- renamed counter to heartbeat
- renamed aggregation to aggregator
- renamed options to config
- reduced code in aggregator selectors

Renamed components props and store reducers
- currentAnswer to latestAnswer
- requestTime to latestRequestTimestamp
- oracleResponses to oracleAnswers
- oracles to oracleList
- updateHeight to latestAnswerTimestamp
- pendingAnswerId to   pendingRoundId

### CODE REVIEW

File: feeds/src/components/vis/graph/GraphTooltip.tsx [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+
+const positionTypes: PositionTypes = {
   oracle: { x: 10, y: 10 },
   contract: { x: 20, y: 20 },
 }
 
-const Tooltip = ({ options, tooltip }) => {
-  const [position, setPosition] = useState({})
+const Tooltip: React.FC<Props> = ({ config, tooltip }) => {
+  const [position, setPosition] = useState<any>({})
```
Comment (HenryNguyen5) [inline feeds/src/components/vis/graph/GraphTooltip.tsx]:
```suggestion
  const [position, setPosition] = useState<PositionTypes>({})
```

File: feeds/src/components/vis/info/Info.tsx [hunk 2]
```diff
@@ -1,85 +1,85 @@
...
 
   const getCurrentResponses = () => {
-    if (!oracleResponse) {
+    if (!oracleAnswers) {
       return '...'
     }
 
-    const responended = oracleResponse.filter((r: any) => {
-      return r.answerId >= pendingAnswerId
+    const responended = oracleAnswers.filter((r: any) => {
```
Comment (HenryNguyen5) [inline feeds/src/components/vis/info/Info.tsx]:
```suggestion
    const responded = oracleAnswers.filter((r: any) => {
```

File: feeds/src/components/vis/info/Info.tsx [hunk 3]
```diff
@@ -1,85 +1,85 @@
...
+      return r.answerId >= pendingRoundId
     })
 
-    return `${responended.length} / ${oracles && oracles.length}`
+    return `${responended.length} / ${oracleList && oracleList.length}`
```
Comment (HenryNguyen5) [inline feeds/src/components/vis/info/Info.tsx]: If `oracleList` is falsey here, wont it return a string that doesnt make sense?
Maybe it should be something like: 
```suggestion
    return oracleList && oracleList.length ? `${responended.length} / ${oracleList.length}` : 'N/A'
```

File: feeds/src/pages/Custom.tsx [hunk 4]
```diff
@@ -13,55 +12,52 @@ interface OwnProps {
...
+  clearState: aggregatorOperations.clearState,
 }
 
-function formatOptions(options: any) {
+function formatConfig(config: any) {
   return {
-    ...options,
-    networkId: uIntFrom(options.networkId),
+    ...config,
+    networkId: uIntFrom(config.networkId || 0),
```
Comment (HenryNguyen5) [inline feeds/src/pages/Custom.tsx]: We should use `??` here and in other statements too

File: feeds/src/state/ducks/aggregator/selectors.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  return result
+})
+
+const latestOraclesState = createSelector(
+  [oracles, oracleAnswers, pendingAnswerId],
+  (list, answers, pendingAnswerId) => {
+    if (!list) return []
+
+    const data = list.map((o: any, id: any) => {
+      const state = answers && answers.find((r: any) => r.sender === o.address)
```
Comment (HenryNguyen5) [inline feeds/src/state/ducks/aggregator/selectors.ts]: Should we do case insensitive comparison here?

File: feeds/src/components/vis/info/Info.tsx (hunk 3 above)
Comment (hellobart) [reply feeds/src/components/vis/info/Info.tsx]: Sounds good. I will use `ellipsis` as its loading indicator. I cannot think about scenario where we don't have any oracles.

File: feeds/src/components/vis/graph/GraphTooltip.tsx (hunk 1 above)
Comment (hellobart) [reply feeds/src/components/vis/graph/GraphTooltip.tsx]: `position` hook is used in inline styles (left, right) so it is required to have a new interface.

File: feeds/src/components/vis/graph/GraphTooltip.tsx [hunk 6]
```diff
@@ -1,14 +1,35 @@
...
+
+const positionTypes: PositionTypes = {
   oracle: { x: 10, y: 10 },
   contract: { x: 20, y: 20 },
 }
 
-const Tooltip = ({ options, tooltip }) => {
-  const [position, setPosition] = useState({})
+const Tooltip: React.FC<Props> = ({ config, tooltip }) => {
+  const [position, setPosition] = useState<PositionStyles | {}>({})
```
Comment (HenryNguyen5) [inline feeds/src/components/vis/graph/GraphTooltip.tsx]:
```suggestion
  const [position, setPosition] = useState<Partial<PositionStyles>>({})
```

File: feeds/src/components/aggregatorVis/AggregatorVis.component.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  minimumAnswers: any
+}
+
+interface DispatchProps {
+  fetchJobId: any
+}
+
+export interface Props extends StateProps, DispatchProps {}
+
+const AggregatorVis: React.FC<Props> = props => <Vis {...props} />
```
Comment (rupurt) [inline feeds/src/components/aggregatorVis/AggregatorVis.component.tsx]: Can you combine `.component` with `.enhanced` e.g. https://github.com/smartcontractkit/chainlink/blob/develop/feeds/src/components/listing/Listing.tsx

File: feeds/src/components/answerHistory/AnswerHistory.component.js [hunk 8]
```diff
@@ -2,14 +2,14 @@ import React, { useEffect, useState, useRef } from 'react'
 import { Icon, Switch, Tooltip } from 'antd'
 import HistoryGraphD3 from './HistoryGraph.d3'
 
-function AnswerHistory({ answerHistory, options }) {
+function AnswerHistory({ answerHistory, config }) {
```
Comment (rupurt) [inline feeds/src/components/answerHistory/AnswerHistory.component.js]: How do you feel about going boy scout on this and converting to TS?

File: feeds/src/components/oracleTable/OracleTable.component.tsx [hunk 9]
```diff
@@ -36,56 +33,46 @@ const OracleTable: React.FC<Props> = ({
...
 
   const columns = [
     {
       title: 'Oracle',
-      dataIndex: 'oracle.name',
+      dataIndex: 'name',
       key: 'name',
-      sorter: (a: any, b: any): number =>
-        a.oracle.name.localeCompare(b && b.oracle && b.oracle.name),
+      sorter: (a: any, b: any): number => a.name.localeCompare(b && b.name),
```
Comment (rupurt) [inline feeds/src/components/oracleTable/OracleTable.component.tsx]: How do you feel about extracting an interface for the columns so we can reduce the type annotations? e.g.

```typescript
interface Column {
   title: string
   dataIndex: string
   key: string
   sorter: (a: any, b: any) => number
   defaultSortOrder?: string
   render?: (timestamp: number) => string
}

const columns: Column[] = [
   {
      title: 'Oracle',
      dataIndex: 'name',
      key: 'name',
      sorter: (a, b) => a.name.localeCompare(b && b.name),
   }
   // ...
]
```

File: feeds/src/components/oracleTable/OracleTable.enhanced.ts [hunk 10]
```diff
@@ -1,19 +1,18 @@
...
+  ethGasPrice: state.aggregator.ethGasPrice,
+  latestOraclesState: aggregatorSelectors.latestOraclesState(state),
 })
 
 const mapDispatchToProps = {
-  fetchEthGasPrice: aggregationOperations.fetchEthGasPrice,
+  fetchEthGasPrice: aggregatorOperations.fetchEthGasPrice,
 }
 
 export default connect(mapStateToProps, mapDispatchToProps)(OracleTable)
```
Comment (rupurt) [inline feeds/src/components/oracleTable/OracleTable.enhanced.ts]: Can you combine `.component` with `.enhanced` e.g. https://github.com/smartcontractkit/chainlink/blob/develop/feeds/src/components/listing/Listing.tsx

File: feeds/src/components/oracleTable/OracleTable.component.tsx (hunk 9 above)
Comment (hellobart) [reply feeds/src/components/oracleTable/OracleTable.component.tsx]: Imported interface straight from antd library

Comment (rupurt) [reply feeds/src/components/oracleTable/OracleTable.component.tsx]: 👍 sweet. Event better 😄

**Pull request merged**